test/test_images/scheduler_target/main.go

```diff
@@ -0,0 +1,56 @@
+package main
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"log"
+	"os"
+
+	cloudevents "github.com/cloudevents/sdk-go"
+	"github.com/kelseyhightower/envconfig"
+)
+
+func main() {
+	client, err := cloudevents.NewDefaultClient()
+	if err != nil {
+		panic(err)
+	}
+
+	r := Receiver{}
+	if err := envconfig.Process("", &r); err != nil {
+		panic(err)
+	}
+
+	if err := client.StartReceiver(context.Background(), r.Receive); err != nil {
+		log.Fatal(err)
+	}
+}
+
+type Receiver struct{}
+
+func (r *Receiver) Receive(event cloudevents.Event) {
```

Comment:
I think it would be better if we create a target image that validates more than just that the event was received. Take a look at the auditlogs target image, and there we are validating more of actual content of the event...
That will make the e2e test pretty robust...

Reply:
+1
I think `SOURCE`, `TYPE`, `SUBJECT` and `TIME` are the parts we can validate, similar as what we have for `AuditLogsSource` - https://github.com/google/knative-gcp/tree/master/test/test_images/auditlogs_target.

Reply:
Do we need to do TYPE? The event isnt a special type like Audit has. Same with SOURCE, it doesn't have a set topic like audit logs does and there is already prefix check in SUBJECT. Also TIME is a timeout not a check.

Reply:
I'm okay with not checking TYPE and SOURCE, if it's hard to do so. Besides we are not checking them either for `PullSubscription` and `Storage` either.

Reply:
hmm why is it hard to check the type or the source? I think it will be useful, so that we don't accidentally break it... 
but I'd rather have this in asap, so that we can mark Scheduler Source to be done. 
We can file an issue to do better assertions in e2e tests in general as a follow up. 
I think I saw some issue that @chizhg filed on how we should improve reading the logs...

Do you agree?

Reply:
I think I can add type since a change just seem to have gone in that actually set it for the event. The Source would be just a prefix check like Subject, so I don't know if it is adding anything. It may be better to leave the discussion for doing better asserts in e2e for a follow up since that is related to a discussion about what we want to be robust against.